imports/lib/schemas/ChatMessage.ts

```diff
@@ -4,12 +4,47 @@ import { BaseCodec, BaseOverrides } from './Base';
 import { Id } from './regexes';
 import { Overrides, buildSchema, inheritSchema } from './typedSchemas';
 
+const MentionBlock = t.type({
+  type: t.literal('mention'),
+  userId: t.string,
+});
+export type ChatMessageMentionNodeType = t.TypeOf<typeof MentionBlock>;
+
+const TextBlock = t.type({
+  text: t.string,
+});
+export type ChatMessageTextNodeType = t.TypeOf<typeof TextBlock>;
+
+const ContentNode = t.union([MentionBlock, TextBlock]);
+export type ChatMessageContentNodeType = t.TypeOf<typeof ContentNode>;
+
+export function nodeIsText(node: ChatMessageContentNodeType): node is ChatMessageTextNodeType {
+  return (node as any).text !== undefined;
+}
+
+export function nodeIsMention(
+  node: ChatMessageContentNodeType
+): node is ChatMessageMentionNodeType {
+  return (node as any).type === 'mention';
+}
+
+const Content = t.type({
+  type: t.literal('message'),
+  children: t.array(t.union([MentionBlock, TextBlock])),
+});
+export type ChatMessageContentType = t.TypeOf<typeof Content>;
+
 const ChatMessageFields = t.type({
   hunt: t.string,
   // The puzzle to which this chat was sent.
   puzzle: t.string,
   // The message body. Plain text.
-  text: t.string,
+  text: t.union([t.string, t.undefined]),
+
+  // The message contents, as an opaque object which has the
+  // schema of Content above
+  content: t.union([t.UnknownRecord, t.undefined]),
```

Comment:
If we're going the opaque object route, it should be possible to export a `ChatMessageType` that uses `ChatMesageContentType` so that we don't actually have to cast.